src/components/shared/PageHeader.vue

```diff
@@ -1,45 +1,197 @@
+import HeaderNavigation from './HeaderNavigation.vue'
+import GenericLink from '../generics/GenericLink.vue'
+import CartIcon from '../../assets/icons/CartIcon.svg'
+import UserIcon from '../../assets/icons/UserIcon.svg'
+import HamburgerMenuIcon from '../../assets/icons/HamburgerMenuIcon.svg'
+import CrossMenuIcon from '../../assets/icons/CrossMenuIcon.svg'
+import HeaderNavigationCollapse from './HeaderNavigationCollapse.vue'
+
+const headerRef = ref(null)
+
+const updateOverlayPosition = () => {
+  nextTick(() => {
+    const header = headerRef.value
+    if (header) {
+      const headerHeight = header.clientHeight
+      const overlay = document.querySelector('.header__overlay')
+      if (overlay) {
+        overlay.style.top = `${headerHeight}px`
+        overlay.style.height = `calc(100vh - ${headerHeight}px)`
+      }
+    }
+  })
+}
+
+let observer // Declare the MutationObserver outside of any specific method to ensure it's accessible throughout the component lifecycle
+
+const updateFollowingBlockPadding = () => {
+  nextTick(() => {
+    const header = headerRef.value
+    if (header) {
+      const nextElement = header.nextElementSibling
+      if (nextElement) {
+        const headerHeight = header.clientHeight
+        nextElement.style.paddingTop = `${headerHeight}px`
+      }
+    }
+  })
+}
+
+onMounted(() => {
+  window.addEventListener('resize', updateMediaFlag)
+  window.addEventListener('resize', updateFollowingBlockPadding)
+  window.addEventListener('resize', updateOverlayPosition)
+  
+  updateFollowingBlockPadding()
+  updateOverlayPosition()
+
+  observer = new MutationObserver(mutations => {
+    mutations.forEach(mutation => {
+      if (mutation.type === 'childList') {
+        updateFollowingBlockPadding()
+      }
+    })
+  })
+
+  observer.observe(document.body, { childList: true, subtree: true })
+})
+
+onUnmounted(() => {
+  window.removeEventListener('resize', updateMediaFlag)
+  window.removeEventListener('resize', updateFollowingBlockPadding)
+  window.removeEventListener('resize', updateOverlayPosition)
+  
+  if (observer) {
+    observer.disconnect()
+  }
+})
 </script>
 
 <template>
-    <header class="header">
+    <header ref="headerRef" class="header">
         <div class="header__container container">
             <HeaderLogo />
-
-            <HeaderNavigation />
-
-            <div class="header__link-wrapper">
-                <GenericLink href="#" containerClass="header__link">
+            <HeaderNavigation v-show="isMenuVisible" />
+            <div class="header__link-wrapper" v-show="isMenuVisible">
+                <GenericLink href="cart" containerClass="header__link">
                     <CartIcon />
                 </GenericLink>
-
                 <GenericLink href="#" containerClass="header__link">
                     <UserIcon />
                 </GenericLink>
             </div>
+            <div class="header__overlay" v-if="!isHamburgerIconVisible" @click="menuToggle"></div>
```

Comment:
Isolate header__overlay and its content in a separate component HeaderModal.vue
(SR)